rdf4j/src/test/java/org/apache/commons/rdf/rdf4j/NativeStoreGraphTest.java

```diff
@@ -49,17 +51,23 @@
  */
 public class NativeStoreGraphTest extends AbstractGraphTest {
 
-	public final class NativeStoreFactory implements RDFTermFactory {
+	public final class NativeStoreRDF implements RDF {
 
-		RDF4JFactory rdf4jFactory = new RDF4JFactory(getRepository().getValueFactory());
+		RDF4J rdf4jFactory = new RDF4J(getRepository().getValueFactory());
 
 		@Override
 		public RDF4JGraph createGraph() {
 			// We re-use the repository connection, but use a different context every time
 			Set<RDF4JBlankNode> context = Collections.singleton(rdf4jFactory.createBlankNode());
 			return rdf4jFactory.asGraph(getRepository(), context);
 		}
-
+		@Override
+		public Dataset createDataset() {
+			throw new UnsupportedOperationException("Can't create more than one Dataset in this test");
```

Comment:
I tried to do that, yes. The problem then was that you couldn't easily see what caused the 30s delay if something was not properly closed. Now that everything is closed properly within the tests it doesn't take many seconds to run all of the tests.